compiler/quilt/test/test_build.py

```diff
@@ -527,8 +527,42 @@ def test_parquet_source_file(self):
                     }
                 }
             }
-        build.build_package_from_contents(None, 'test', 'fromparquet', '.', build_contents)
+        build.build_package_from_contents(None, 'test', 'fromparquet', [], '.', build_contents)
         pkg = command.load('test/fromparquet')
         assert df.equals(pkg.df()) # pylint:disable=E1101
 
     #TODO: Add test for checks on a parquet-sourced dataframe
+
+    def test_subpackage(self):
+        mydir = pathlib.Path(os.path.dirname(__file__))
+
+        command.build('test/foo')
+
+        df = DataFrame(dict(a=[1, 2, 3]))
+        arr = np.array([4, 5, 6])
+        path = str(mydir / 'build_simple.yml')
+
+        command.build('test/foo/empty')
+        command.build('test/foo/df', df)
+        command.build('test/foo/arr', arr)
+        command.build('test/foo/file', path)  # Adds as a plain file
+        command.build('test/foo/stuff', path, build_file=True)  # Builds a subpackage
```

Comment:
Nice-to-have, but not required: instead do `data_file=True` if the user wants to "add as a plain file" so that the signature of `quilt.build` is consistent across both packages and subpackages.